app/Services/InviteService.php

```diff
@@ -2,32 +2,66 @@
 
 namespace App\Services;
 
+use Firebase\JWT\JWT;
 use App\Models\Invite;
 use App\Mail\InviteMail;
 use Illuminate\Support\Facades\Mail;
+use App\Constants\Invite as InviteConstant;
+use App\Http\Resources\UserResource;
+use App\Http\Resources\InviteResource;
 
 class InviteService
 {
-    public function generateToken(): string
+    public function generateToken(array $user): string
     {
-        return bcrypt(now()->addDays(5)->toDateTimeLocalString());
+        $issuedAt = time();
+
+        $expirationTime = $issuedAt . InviteConstant::DEFAULT_EXIPIRE_AT;
+
+        error_log($expirationTime);
+
+        $payload = array(
+            'name' => $user['name'],
+            'email' => $user['email'],
+            'iat' => $issuedAt,
+            'exp' => $expirationTime,
+        );
+
+        $token = JWT::encode(
+            $payload,
+            config('jwt.secret_key'),
+            config('jwt.algorithm')
+        );
+
+        return $token;
     }
 
-    public function processInvite(array $validatedInviteUser): mixed
+    public function processInvite(array $validatedInviteUser)
     {
-        $token = $this->generateToken();
+        $token = $this->generateToken([
+            'name' => $validatedInviteUser['name'],
+            'email' => $validatedInviteUser['email']
+        ]);
 
         $url = config('frontend.url') . '/register/' . $token . '?email=' . $validatedInviteUser['email'] . '&name=' . urlencode($validatedInviteUser['name']);
```

Comment:
@sabinshrestha8 Better to create a helper function within same class or creating trait, if this is reusable